src/components/item/Container.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import { Item } from './Presenter';
+import axios from 'axios';
+
+export default class ItemContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      category: "",
+      description: "",
+      endDate: "",
+      id: props.match.params.id,
+      image: "",
+      loading: true,
+      price: "",
+      title: "",
+    }
+  }
+
+  fetchItem = async () => {
+    const self = this;
+    await axios.get(`/api/items/${self.state.id}`)
+      .then(function (response) {
+        const item = response.data.data;
+        self.setState({
+          category: item.category,
+          description: item.description,
+          endDate: item.end_date,
+          image: item.image,
+          loading: false,
+          price: item.price,
+          title: item.title,
+        });
+      })
+      .catch(function (error) {
+        console.log(error);
+        return error.message;
+      });
+  }
+
+  componentDidMount() {
+    this.fetchItem(this);
+  }
+
+  render() {
+    return (
+      !this.state.loading &&
+      <Item props={this.state} />
```

Comment:
At this current moment, it doesn't seem like we need state / class component. Since its a simple fetch and pass of data into a <Item /> component, maybe we can use a functional component + React Hooks ?